packages/animations/lib/src/modal.dart

```diff
@@ -0,0 +1,208 @@
+
+  @override
+  Duration get reverseTransitionDuration => _reverseTransitionDuration;
+  final Duration _reverseTransitionDuration;
+
+  /// The primary contents of the modal.
+  final WidgetBuilder builder;
+
+  @override
+  Widget buildPage(
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+  ) {
+    final ThemeData theme = Theme.of(context);
+    return Semantics(
+      child: SafeArea(
+        child: Builder(
+          builder: (BuildContext context) {
+            final Widget child = Builder(builder: builder);
+            return theme != null ? Theme(data: theme, child: child) : child;
+          },
+        ),
+      ),
+      scopesRoute: true,
+      explicitChildNodes: true,
+    );
+  }
+
+  @override
+  Widget buildTransitions(
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return _transitionBuilder(
+      context,
+      animation,
+      secondaryAnimation,
+      child,
+    );
+  }
+
+  final _ModalTransitionBuilder _transitionBuilder;
+}
+
+/// A configuration object containing the properties needed to implement a
+/// modal route.
+///
+/// The `barrierDismissible` argument is used to determine whether this route
+/// can be dismissed by tapping the modal barrier. This argument defaults
+/// to true. If `barrierDismissible` is true, a non-null `barrierLabel` must be
+/// provided.
+///
+/// The `barrierLabel` argument is the semantic label used for a dismissible
+/// barrier. This argument defaults to "Dismiss".
+abstract class ModalConfiguration {
+  /// Creates a modal configuration object that provides the necessary
+  /// properties to implement a modal route.
+  ModalConfiguration({
+    this.barrierColor,
+    this.barrierDismissible,
+    this.barrierLabel,
+    this.transitionDuration,
+    this.reverseTransitionDuration,
+  });
+
+  /// The color to use for the modal barrier. If this is null, the barrier will
+  /// be transparent.
+  final Color barrierColor;
+
+  /// Whether you can dismiss this route by tapping the modal barrier.
+  final bool barrierDismissible;
+
+  /// The semantic label used for a dismissible barrier.
+  final String barrierLabel;
+
+  /// A builder that defines how the route arrives on and leaves the screen.
+  ///
+  /// The [buildTransitions] method is typically used to define transitions
+  /// that animate the new topmost route's comings and goings. When the
+  /// [Navigator] pushes a route on the top of its stack, the new route's
+  /// primary [animation] runs from 0.0 to 1.0. When the [Navigator] pops the
+  /// topmost route, e.g. because the use pressed the back button, the
+  /// primary animation runs from 1.0 to 0.0.
+  Widget transitionBuilder(
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  );
+
+  /// The duration of the transition running forwards.
```

Comment:
Move these two properties above the method to have all properties grouped together?